### DISCUSSION TITLE
Hook to validate a photo

### DISCUSSION THREAD

**Discussion by patryk-kawalarz** [directus/directus#8768] [Archive]
In my collection I have a field "profile_photo" and I want to check it's size and dimensions and show alert if it's not correct. I wanted to create a hook, but I can't get those informations about a file. Can you help me?

**Comment by joselcvarela** [comment]
Hello @patryk-kawalarz 
As I mentioned here https://github.com/directus/directus/discussions/8704#discussioncomment-1462246, you can rely on `files.upload` custom hook to do this.
Inside that hook you will have access to `payload.filename_disk` so you can use `fs.stats` to check the size of the file and use `image-size` package, for example, to check the image dimensions

**Reply by patryk-kawalarz** [reply]
Thank you! One more question: can I set this field as invalid to block saving an item?

**Reply by joselcvarela** [reply]
I am afraid you can't.
This hook is invoked after the upload and the creation of the file.
But, you can do your checks and if it fails you can remove file, remove entry on directus_files and throw an error.
It's a bit hacky but it should work.

I could suggest to implement `files.upload.before` but I think that's not actually possible because the upload stream must be saved somewhere so you can have access to the entire file.
